server/tests/document.spec.js

```diff
@@ -0,0 +1,247 @@
+      returning: true })
+      .then((createdRoles) => {
+        adminRole = createdRoles[0];
+        regularRole = createdRoles[1];
+        adminUserParam.RoleId = adminRole.id;
+        regularUserParam.RoleId = regularRole.id;
+
+        request.post('/users')
+          .send(adminUserParam)
+          .end((error, response) => {
+            adminUser = response.body.user;
+            adminToken = response.body.token;
+            documentOne.UserId = adminUser.id;
+            documentOne.RoleId = adminRole.id;
+
+            request.post('/users')
+              .send(regularUserParam)
+              .end((err, res) => {
+                regularUser = res.body.user;
+                regularToken = res.body.token;
+                documentTwo.UserId = regularUser.id;
+                documentTwo.RoleId = regularRole.id;
+                done();
+              });
+          });
+      });
+  });
+
+  after(() => model.sequelize.sync({ force: true }));
+
+  describe('Post (/documents) - Create document', () => {
+    it('Should have a published date defined', (done) => {
+      request.post('/documents')
+        .set({ 'x-access-token': adminToken })
+        .send(documentOne)
+        .expect(201)
+        .end((err, res) => {
+          document = res.body;
+          expect(res.body).to.have.property('createdAt');
+          expect(res.body.createdAt).not.to.equal(null);
+          done();
+        });
+    });
+
+    it('Should have valid attributes', (done) => {
+      expect(document).to.have.property('title');
+      expect(document).to.have.property('content');
+      done();
+    });
+
+    it('Should create a document with a unique title', (done) => {
+      request.post('/documents')
+        .set({ 'x-access-token': adminToken })
+        .send(documentOne)
+        .end((err, res) => { // eslint-disable-line no-unused-vars
+          request.post('/documents')
+            .set({ 'x-access-token': adminToken })
+            .send(documentOne)
+            .end((err, res) => {
+              expect(res.body.errors[0].message).to.equal('title must be unique');
+              done();
+            });
+        });
+    });
+
+    it('Should ensure that document has an owner', (done) => {
+      request.post('/documents')
+        .set({ 'x-access-token': regularToken })
+        .send(documentTwo)
+        .expect(201)
+        .end((err, res) => {
+          document = res.body;
+          expect(res.body.UserId).to.equal(regularUser.id);
+          done();
+        });
+    });
+
+    it('Should ensure that document has a role that can access it', (done) => {
+      expect(document.RoleId).to.equal(regularRole.id);
+      done();
+    });
+
+    it('Should ensure that document cannot be created if title is lacking', (done) => {
+      const nullTitleDoc = { title: null, content: 'content' };
+      request.post('/documents')
+        .set({ 'x-access-token': adminToken })
+        .send(nullTitleDoc)
+        .expect(201)
```

Comment:
Why are you expecting status `201` for a document that was not created? Is this test passing? If it is, are you returning status code 201 from the controller or this expect is actually redundant